pkg/vault/contracts/VaultExtension.sol

```diff
@@ -284,7 +287,8 @@ contract VaultExtension is IVaultExtension, VaultCommon, Proxy {
                 // input amount (EXACT_OUT).
                 if (
                     hookFlags.enableHookAdjustedAmounts &&
-                    params.liquidityManagement.disableUnbalancedLiquidity == false
+                    (params.liquidityManagement.disableAddLiquidityUnbalanced == false ||
+                        params.liquidityManagement.disableRemoveLiquidityUnbalanced == false)
```

Comment:
hm.
The problem is that we want to enable this when the adjusted amount is only for one operation.

E.g. we can enable after exit adjusted amounts with unbalanced adds. Which means that we'd need to also split `hookAdjustedAmounts` flag as well, unless I'm missing something.

Reply:
We could have hookAdjustedAmountsIn and hookAdjustedAmountsOut. We also check this on swaps, though... do we need three?